component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtil.java

```diff
@@ -785,6 +787,50 @@ public static boolean isLocalUser(AuthenticationContext context) {
         return false;
     }
 
+    /**
+     * Get decrypted stored secret key for federated users.
+     *
+     * @param context Authentication context.
+     * @param userId Federated unique user id.
+     * @return Decrypted stored secret key
+     * @throws TOTPException if an error occurred while decrypting the stored secret key.
+     */
+    public static String getSecretKey(AuthenticationContext context, String userId) throws TOTPException {
+
+        try{
+            String storedSecretKey;
+            if (context.getProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL) != null) {
+                storedSecretKey = context.getProperty(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL).toString();
+            } else {
+                storedSecretKey = DAOFactory.getInstance().getTOTPSecretKeyDAO().
+                        getTOTPSecretKeyOfFederatedUser(userId);
+            }
+            return decrypt(storedSecretKey);
+        } catch (CryptoException e) {
+            throw new TOTPException("Error while decrypting the key", e);
+        }
+    }
+
+    /**
+     * Returns AuthenticatedUser object from context.
+     *
+     * @param context AuthenticationContext.
+     * @return AuthenticatedUser
+     */
+    public static AuthenticatedUser getAuthenticatedUser(AuthenticationContext context) {
+
+        AuthenticatedUser authenticatedUser = null;
+        Map<Integer, StepConfig> stepConfigMap = context.getSequenceConfig().getStepMap();
```

Comment:
can context / context.getSequenceConfig() become null here? if so handle null checks.

Reply:
At this point context / context.getSequenceConfig() can't be null here. Handle null check for stepConfigMap here.